src/main/kotlin/org/rust/ide/typing/assist/CommaFixer.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.typing.assist
+
+import com.intellij.lang.SmartEnterProcessorWithFixers
+import com.intellij.openapi.editor.Editor
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.RsCallExpr
+import org.rust.lang.core.psi.RsMacroExpr
+import org.rust.lang.core.psi.RsMatchArm
+
+/**
+ * Fixer that adds missing comma at the end of statements.
+ */
+class CommaFixer : SmartEnterProcessorWithFixers.Fixer<RsSmartEnterProcessor>() {
+    override fun apply(editor: Editor, processor: RsSmartEnterProcessor, element: PsiElement) {
+        if (element is RsMatchArm) {
+            val lastElement = element.lastChild
+            if ((lastElement is RsCallExpr || lastElement is RsMacroExpr) && element.lastChild.text != ",")
```

Comment:
Why do we need `(lastElement is RsCallExpr || lastElement is RsMacroExpr)` condition?
What if I write some expression like `1` or `"foo".as_bytes()`?

Reply:
Is to prevent this from happening:

```rust
fn main() {
    let a = 1;

    match a {
        1 => <caret>
    };
}
```

I don't think this situation should add commas. What do you think? @Undin